### PR TITLE
adds project funder to user that funds

### PR DESCRIPTION
**Pull request by scmountain**
Rspec 73 - 1 - 2
Fails on "1) a registered user in the create project page can create project, gain project owner role and add a contributor
     Failure/Error: <%= image_tag @registered_user.avatar_url, class: "image-size" %>

     ActionView::Template::Error:
       undefined method `avatar_url' for nil:NilClass"

### CODE REVIEW

File: app/views/user_funded_projects/new.html.erb [hunk 1]
```diff
@@ -11,6 +11,8 @@
 
         <%= f.label :credit_card_number %>
         <%= f.text_field :credit_card_number %>
+        <%= f.submit "Fund" %>
+        <%#= link_to "Fund", project_user_funded_projects_path(@current_project), class: "btn right btn-color" %>
```
Comment (cews7) [inline app/views/user_funded_projects/new.html.erb]: @scmountain did we want to take out line 15?

**Comment by cews7** [comment]
@scmountain I think I can fix the failing test.

File: app/controllers/user_funded_projects_controller.rb [hunk 2]
```diff
@@ -5,4 +5,20 @@ def new
     @project = Project.find(params[:project_id])
   end
 
+  def create
+    @user = current_user
+    @project = Project.find(params[:project_id])
+    fund = @user.user_funded_projects.create(project_params)
+    @role = Role.find_or_create_by(name: "project_funder")
+    @user.roles << @role
+    @project.funders << @user
```
Comment (cews7) [inline app/controllers/user_funded_projects_controller.rb]: @scmountain for the funders method, what table supplies that relationship?

File: spec/features/project_funder_exists_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    fill_in "user_funded_project[credit_card_number]", with: "123412342341234"
+
+    click_on "Fund"
+
+    expect(current_path).to eq("/projects/#{project.slug}")
+    expect(user.roles.count).to eq(2)
+    expect(user.roles.first.name).to eq("registered_user")
+    expect(user.roles.last.name).to eq("project_funder")
+  end
+end
```
Comment (cews7) [inline spec/features/project_funder_exists_spec.rb]: 🔥

File: app/controllers/user_funded_projects_controller.rb [hunk 4]
```diff
@@ -5,4 +5,20 @@ def new
...
+    flash.now[:notice] = "You just funded #{@project.name} for #{fund.amount}!"
+    redirect_to project_path(@project.slug)
+  end
+
+  private
+
+  def project_params
+    params.require(:user_funded_project).permit(:amount, :credit_card_number).merge(project_id: params["project_id"])
```
Comment (CjMoore) [inline app/controllers/user_funded_projects_controller.rb]: whats going on with this merge?

File: app/controllers/user_funded_projects_controller.rb (hunk 2 above)
Comment (CjMoore) [inline app/controllers/user_funded_projects_controller.rb]: If you look at the project model there is a relationship that looks like 'has_many: :funders, through: :user_funded_projects, source: :users'

Comment (CjMoore) [inline app/controllers/user_funded_projects_controller.rb]: Although I'm not sure if this line is necessary as we establish that relationship on line 11.

File: spec/features/project_funder_exists_spec.rb (hunk 3 above)
Comment (CjMoore) [inline spec/features/project_funder_exists_spec.rb]: I think the other part of this card was making sure that when a user who has funded a project visits that project show, it displays how much that user has funded. We might want to add that expectation here to make sure that amount is displayed.

Comment (scmountain) [inline spec/features/project_funder_exists_spec.rb]: it was a different card for that functionality

Comment (CjMoore) [inline spec/features/project_funder_exists_spec.rb]: got it.

**Pull request merged**